src/app/ReadClient.cpp

```diff
@@ -957,34 +972,32 @@ void ReadClient::OnResubscribeTimerCallback(System::Layer * apSystemLayer, void
     _this->mNumRetries++;
 }
 
-bool ReadClient::ResubscribeIfNeeded()
+void ReadClient::ResubscribeIfNeeded(bool & aResubscribe, uint32_t & aNextResubscribeIntervalMsec)
```

Comment:
This method needs API documentation. More importantly, this behavior of only setting it to `true` if conditions apply, but not actually setting it to false if not is not a great API pattern.

Reply:
agree, updated API documentation, and also set aResubscribe, aNextResubscribeIntervalMsec in the beginning.